src/modules/discovery/kube_state_metrics.py

```diff
@@ -0,0 +1,42 @@
+
+import logging
+import subprocess 
+import json
+
+from ...core.types import Discovery
+from ...core.events import handler
+from ...core.events.types import HuntStarted, Event
+
+
+class KubeStateMetricsEvent(Event):
+    """Add-on agent to generate and expose cluster-level metrics."""
+    def __init__(self, version):
+        self.version = version
+
+# Will be triggered on start of every hunt 
+@handler.subscribe(HuntStarted)
+class KubeStateMetricsDiscovery(Discovery):
+    """kube-state-metrics version Discovery
+    Checks the version of kube-state-metrics part of CVE-2019-17110 vulnerability check
+    """
+    def __init__(self, event):
+        self.event = event
+
+    def get_kube_state_metrics_version(self):
+        version = None
+        try:
+            # kube-state-metrics affected versions: v1.7.0 and v1.7.1
+            kubectl_cmd = "kubectl get deployment -n kube-system kube-state-metrics -o yaml | grep image:"
```

Comment:
I don't believe we have kubectl in the kube-hunter docker image. is there another way to determine the version?

Reply:
I am not sure about this one, but let me check first.

Reply:
Further than not having kubectl binary, we don't have any kubeconfig context.
Discovery of services such as kube-state-metrics should be done via subscribing to `OpenPortEvent` giving a predicate to filter service port (refer to [KubeDashboard discovery](https://github.com/aquasecurity/kube-hunter/blob/master/src/modules/discovery/dashboard.py#L16)).
Once you find a service endpoint, you can call an API to fetch its version.

Reply:
I will check it out, thanks for the guidance @iYehuda